src/controllers/uploads.js

```diff
@@ -47,4 +50,47 @@ export default {
       next(error);
     }
   },
+
+  deleteAvatarImage: async (req, res, next) => {
+    const bucket = process.env.AWS_S3_BUCKET;
+    try {
+      const { url } = req.body;
+      const imageUrl = new URL(url);
+      if (
+        imageUrl.host != `${bucket}.${AWS_S3_DOMAIN}` ||
+        !imageUrl.pathname.startsWith(`/${KEY_PATH}`)
+      ) {
+        respondWithError(res, {}, httpStatus.BAD_REQUEST);
+      } else {
+        // Check that the url is not used in the users db
+        const avatarExists = await User.findOne({
+          where: {
+            avatarUrl: url,
+          },
+        });
+        if (!avatarExists) {
+          const pathSplit = imageUrl.pathname.split('/');
+          const fileName = pathSplit[pathSplit.length - 1];
+          const key = `${KEY_PATH}${fileName}`;
+          const params = {
+            Bucket: bucket, // The name of the bucket.
+            Key: key, // The name of the object.
+          };
+          const results = await s3.send(new DeleteObjectCommand(params));
+          respondWithSuccess(
+            res,
+            { avatarUrlIsUsed: false },
+            results.$metadata.httpStatusCode,
+          );
+        } else {
+          respondWithSuccess(
+            res,
+            { avatarUrlIsUsed: true }, // the avatar image cannot be deleted because is used by an user entry
```

Comment:
I am not really sure that we should return a success if the url is used. Maybe it should be best to return an error like https://developer.mozilla.org/en-US/docs/Web/HTTP/Status/422

Reply:
That makes sense